stdlib/@tests/test_cases/check_copy.py

```diff
@@ -19,3 +20,20 @@ def __replace__(self, val: int) -> Self:
     obj = ReplaceableClass(42)
     cpy = copy.replace(obj, val=23)
     assert_type(cpy, ReplaceableClass)
+
+
+_T_co = TypeVar("_T_co", covariant=True)
+
+
+class Box(Generic[_T_co]):
+    def __init__(self, value: _T_co, /) -> None:
+        self.value = value
+
+    def __replace__(self, value: Any) -> Box[Any]:
+        return Box(value)
```

Comment:
Would it make sense to use some other type than just `Any`, just to be more obvious what this does? `Any` is always a bit dangerous, since it just turns type checking off, so it's a bit hard to "prove" anything with it.

Maybe use this here and then `assert_type(box2, Box[str])` below?


```suggestion
    def __replace__(self, value: str) -> Box[str]:
        return Box(value)
```

Reply:
First I tried it with `(self, value: T) -> Box[T]`, expecting it to be inferred as `Box[Any]` that way, but I ran into a pyright bug ([pyright-play](https://pyright-play.net/?pythonVersion=3.14&strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoCCKcANFAArgxgDGYANmcKgIb0BQ7AAsym%2BzXosAzsKgAhMAA8A2gBUAugC52UNVAD6ANzYBXAKZKoczuqgATfcE0bUSGBo0AKYfvrAyO%2BgaNyyAegBKKABaAD4oADkwFENVM3VXdwA6bT19KABeKC8DU3VLa0cQfQQhGn1HGQA1RRc3Dxz0o1rg8IlpGsUVBPUSmF0QFA6pJ1z9QM5BETEAYRYUACVS8v0ZRbrKMGo6ekCegqsbErKWCsd69wCyACoWEDRhIyJSKBubgGsAd3vH5%2BI2hENkZkqDOIUoCdVus6gAjIzzJYrM5rDYKa5vT4-B5PQgA0JAuQHNT9QbDWGpDRQlEXd7fX7CSbsEpafRsDTwBD6JzUipOSSjAAsACZAp50pkAETCBAsCCSwKTADExkQGTA1klvO5AqcIrFTW8%2BilMrlCslmDEkoFXS4xWR5w0Ckl7CAA)).

But yea I agree that it's best to avoid `Any`; I'll apply your suggestion.